assets/js/components/lessonList.js

```diff
@@ -282,18 +299,18 @@
 
     checkTrialByType: function (type) {
       var $tree = $('#js-course-tree-ajax');
-      var $enrollment = $tree.data('enrollment');
+      var enrollment = $tree.data('enrollment');
       var constrains_name = 'trial_' + type + '_ids';
 
-      var payment_method = $enrollment.payment_method;
-      var constrains_tree = $tree.data('school-product').trial_constrains;
-      var constrains = constrains_tree[payment_method];
+      var payment_method = enrollment && enrollment.payment_method;
+      var constrains_tree = $tree.data('school-product') && $tree.data('school-product').trial_constrains;
+      var constrains = constrains_tree && payment_method && constrains_tree[payment_method];
 
       if (constrains && !constrains.hasOwnProperty(constrains_name)) {
         return false;
       }
 
-      if (!$enrollment['on_trial?']) {
+      if (enrollment && !enrollment['on_trial?']) {
```

Comment:
@mandado outra parada, nessa função só to vendo retornar false, quem retorna true?